src/service/debtor-service.ts

```diff
@@ -365,7 +367,14 @@ export default class DebtorService {
     // Get all transfers that have a fine or waived fine
     const transfers = await Transfer.find({
       relations: ['fine', 'fine.transfer',  'waivedFines', 'waivedFines.waivedTransfer'],
-      where: [{ fine: true }, { waivedFines: true }],
+      where: [{ fine: true, createdAt: Raw(
+        (alias) => `${alias} >= :fromDate AND ${alias} < :tillDate`,
+        { fromDate: toMySQLString(fromDate), tillDate: toMySQLString(toDate) },
+      ) },
+      { waivedFines: true, createdAt: Raw(
+        (alias) => `${alias} >= :fromDate AND ${alias} < :tillDate`,
+        { fromDate: toMySQLString(fromDate), tillDate: toMySQLString(toDate) },
+      ) }],
```

Comment:
This already seems to have been merged, but instead of writing these queries yourself, please use the `createFilterWhereDate()` function which already implements this for you.